src/filesystem.hpp

```diff
@@ -121,14 +121,13 @@ static const blacklist_pattern_list default_blacklist{
 };
 
 /**
- * Populates 'files' with all the files and
- * 'dirs' with all the directories in dir.
- * If files or dirs are nullptr they will not be used.
- *
- * mode: determines whether the entire path or just the filename is retrieved.
- * filter: determines if we skip images and sounds directories
- * reorder: triggers the special handling of _main.cfg and _final.cfg
- * checksum: can be used to store checksum info
+ * @param dir The directory to examine.
+ * @param files The files in @a dir. Won't be used if nullptr.
+ * @param dirs The directories in @a dir. Won't be used if nullptr.
+ * @param mode Determines whether the entire path or just the filename is retrieved.
+ * @param filter Determines if we skip images and sounds directories.
+ * @param reorder Triggers the special handling of _main.cfg and _final.cfg.
+ * @param checksum Can be used to store checksum info.
```

Comment:
This updated doc comment seems to be simultaneously an upgrade and a downgrade. It's an upgrade because it uses proper doxygen syntax, but a downgraded because it loses the genera description.

Also, I think you should mark the output parameters as such.

```suggestion
 * Get a list of all files and/or directories in a given directory.
 * 
 * @param dir The directory to examine.
 * @param[out] files The files in @a dir. Won't be used if nullptr.
 * @param[out] dirs The directories in @a dir. Won't be used if nullptr.
 * @param mode Determines whether the entire path or just the filename is retrieved.
 * @param filter Determines if we skip images and sounds directories.
 * @param reorder Triggers the special handling of _main.cfg and _final.cfg.
 * @param[out] checksum Can be used to store checksum info.
```

Reply:
The same information is present in the `@param` descriptions, unless I'm missing something.  I'll add [out] though.

Reply:
I just think there should be a summary line on all functions. Maybe an exception can be made for extremely obvious cases like `copy_file`, but this definitely isn't one of those.